superset/assets/src/visualizations/leaflet_map.js

```diff
@@ -372,6 +373,54 @@ function leafletmap(slice, payload) {
         renderBasicMap();
         renderPolygonLayer();
         addColumnsDropdownToMap();
+        addHeatMapColorLegend();
+    }
+
+    function addHeatMapColorLegend(){
+      var colname  = getSelectedColorColumn()
+      var col = colorCols[colname];
+      var minValue = col['operator'];
+      var maxvalue = col['sqlExpression'];
+      var legends = splitRangeIntoEqualParts(parseFloat(minValue), parseFloat(maxvalue), 10);
+
+      var legend = L.control({ position: 'topleft' });
+      legend.onAdd = () => {
+        var div
+        if($('#heat-map-color-legend').length > 0){
+          div = $('#heat-map-color-legend')[0];
+        } else {
+          div = L.DomUtil.create('div');
+          div.setAttribute('id', 'heat-map-color-legend')
+
+          //to show legend horizontally, un-comment below line
+          //div.classList.add("horizontal");
+
+        }
+        div.innerHTML = getHeatMapColorLegendHTML(colname, legends);
+        return div;
+      };
+      legend.addTo(mapInstance);
+    }
+
+    function getHeatMapColorLegendHTML(colname, legends){
+      var html = "<div class='percentage-value'>0%</div>"
```

Comment:
rename it to  getLegendComponent or getMapLegend

Reply:
done